src/v2/Apps/Fair/Routes/FairExhibitors.tsx

```diff
@@ -1,219 +1,60 @@
-      }
-    )
-
-    if (filtersHaveUpdated) {
-      fetchResults()
-    }
-  }, [context.filters])
-
-  function fetchResults() {
-    setIsLoading(true)
-
-    const relayParams = {
-      id: fair.slug,
-      first: PAGE_SIZE,
-    }
-
-    const relayRefetchVariables = {
-      ...relayParams,
-      ...context.filters,
-    }
-
-    relay.refetch(relayRefetchVariables, null, error => {
-      if (error) {
-        logger.error(error)
-      }
-
-      setIsLoading(false)
-    })
-  }
-
-  function loadPage(page) {
-    context.setFilter("page", page)
-  }
-
-  function loadNext() {
-    if (fair.exhibitors?.pageInfo.hasNextPage) {
-      loadPage(context?.filters?.page! + 1)
-    }
-  }
+const FairExhibitors: React.FC<FairExhibitorsProps> = ({ fair }) => {
+  const { isEnteredView, Waypoint } = useLazyLoadComponent()
 
   return (
     <>
-      <Box id="jump--ExhibitorsFilter" />
-
-      <Media at="xs">
-        <Sticky>
-          {({ stuck }) => {
-            return (
-              <Flex
-                justifyContent="flex-end"
-                py={1}
-                {...(stuck
-                  ? {
-                      px: 2,
-                      borderBottom: "1px solid",
-                      borderColor: "black10",
-                    }
-                  : {})}
-              >
-                <FairExhibitorSortFilter />
-              </Flex>
-            )
-          }}
-        </Sticky>
-      </Media>
-
-      <Media greaterThan="xs">
-        <Flex justifyContent="flex-end">
-          <FairExhibitorSortFilter />
-        </Flex>
-      </Media>
-
-      <Spacer mt={4} />
+      <Waypoint />
 
-      <LoadingArea isLoading={isLoading}>
-        {fair.exhibitors?.edges!.map((edge, index) => {
-          const show = edge?.node!
-          if (show?.counts?.artworks === 0 || !show?.partner) {
-            // Skip rendering of booths without artworks
-            return null
-          }
-
-          return (
-            <Box my={6} key={index}>
-              <FairExhibitorRail key={show.id} show={show} />
-            </Box>
-          )
-        })}
-      </LoadingArea>
-
-      <Spacer mt={4} />
+      {fair.exhibitorsGroupedByName?.map(exhibitorsGroup => {
+        if (!exhibitorsGroup?.exhibitors?.length || !exhibitorsGroup.letter) {
+          return null
+        }
 
-      <Pagination
-        hasNextPage={hasNextPage}
-        pageCursors={pageCursors}
-        onClick={(_cursor, page) => loadPage(page)}
-        onNext={() => loadNext()}
-        scrollTo="#jump--ExhibitorsFilter"
-      />
+        const partnerIds = exhibitorsGroup.exhibitors.map(
+          exhibitor => exhibitor?.partnerID || ""
+        )
+
+        return (
+          <React.Fragment key={exhibitorsGroup.letter}>
+            <Text variant="lg" my={4}>
+              {exhibitorsGroup.letter}
+            </Text>
+            {isEnteredView ? (
+              <FairExhibitorsGroup ids={partnerIds} />
```

Comment:
**question**: If Metaphysics supported a `partner` field under the `FairExhibitor` field, would we be able to fetch all of the data in the top-level query instead of fanning out to multiple QueryRenders? That might be worth adding in so we can simplify the data fetching here.

Reply:
`FairExhibitor` type includes only partner `id` and `name`. I'm not sure if we can fetch all partners info in the top-level query.

Reply:
It doesn't now - but I believe Metaphysics could be updated to expose a `partner` field under `FairOganizer` that would allow you to query for everything in one QueryRenderer.

Something like:

```graphql
{
  fair(id: "the-armory-show-2020") {
    name
    exhibitorsGroupedByName {
      letter
      exhibitors {
        partner {
          name
          href
          profile {
            icon {
              cropped(width: 50, height: 50) {
                src
                srcSet
              }
            }
          }
        }
      }
    }
  }
}
```

What that be preferable to breaking out the groups and querying separately?

Reply:
Yes, I agree, maybe it will be better to update metaphysics and add `Partner`  type to `FairExhibitors`.
But for now it implemented like this 🙂